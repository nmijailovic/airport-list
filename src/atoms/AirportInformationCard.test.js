import React from 'react';
import renderer from 'react-test-renderer';

import AirportDetailsCard from './AirportInformationCard';

// Performing a shallow render unit test
// Could be expanded to traverse the DOM tree and make sure Component is showing correct details as per https://reactjs.org/docs/test-renderer.html
describe('airport information card', () => {
  let item = {};
  let component;
  let testInstance;

  beforeEach(() => {
    item = {
      airportCode: 'AAA',
      internationalAirport: false,
      domesticAirport: false,
      regionalAirport: false,
      onlineIndicator: false,
      eticketableAirport: false,
      location: {
        aboveSeaLevel: -99999,
        latitude: 17.25,
        latitudeRadius: -0.304,
        longitude: 145.3,
        longitudeRadius: -2.5395,
        latitudeDirection: 'S',
        longitudeDirection: 'W'
      },
      airportName: 'Anaa',
      city: {
        cityCode: 'AAA',
        cityName: 'Anaa',
        timeZoneName: 'Pacific/Tahiti'
      },
      country: {
        countryCode: 'PF',
        countryName: 'French Polynesia'
      },
      region: {
        regionCode: 'SP',
        regionName: 'South Pacific'
      }
    };

    component = renderer.create(<AirportDetailsCard item={item} />);
    testInstance = component.root;
  });

  it('has correct structure', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has correct value for region', () => {

    const spans = testInstance.findAll((el) => {
      return el.type === 'span'
    });

    expect(spans[0].children).toEqual(['South Pacific']);
  });

});
