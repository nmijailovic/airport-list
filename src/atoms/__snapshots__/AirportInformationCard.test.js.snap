// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`airport information card has correct structure 1`] = `
<div
  className="card"
>
  <header
    className="card-header"
  >
    <p
      className="card-header-title"
    >
      Anaa
    </p>
  </header>
  <div
    className="card-content"
  >
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           International
        </div>
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           Domestic
        </div>
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           Regional
        </div>
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           ETickets
        </div>
      </div>
      <div
        className="column"
      >
        <div>
          <label>
            Region:
          </label>
          <span>
            South Pacific
          </span>
          <div>
            <label>
              Country:
            </label>
            <span>
              French Polynesia
            </span>
          </div>
          <div>
            <label>
              Time Zone:
            </label>
            <span>
              Pacific/Tahiti
            </span>
          </div>
          <div>
            <label>
              Latitude:
            </label>
            <span>
              -17.25
            </span>
          </div>
          <div>
            <label>
              Longitude:
            </label>
            <span>
              -145.3
            </span>
          </div>
          <div
            className="is-link"
          >
            <a
              href="https://www.google.com/maps/@-17.25,-145.3,12z"
            >
              View on Map
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
